frontend/model/notifications/mainNotificationsMixin.js

```diff
@@ -0,0 +1,204 @@
+        compareISOTimestamps(now, incomeDetailsLastUpdatedDate) > 6 * MONTHS_MILLIS &&
+        !myNotificationHas(item => item.type === 'INCOME_DETAILS_OLD' && item.data.lastUpdatedDate === incomeDetailsLastUpdatedDate)
+    },
+    emit ({ rootGetters }) {
+      sbp('gi.notifications/emit', 'INCOME_DETAILS_OLD', {
+        createdDate: new Date().toISOString(),
+        months: 6,
+        lastUpdatedDate: rootGetters.ourGroupProfile.incomeDetailsLastUpdatedDate
+      })
+    }
+  }
+]
+
+const periodicNotificationEntries = [
+  {
+    type: PERIODIC_NOTIFICATION_TYPE.MIN1,
+    notificationData: {
+      stateKey: 'nearDistributionEnd',
+      emitCondition ({ rootGetters }) {
+        const currentPeriod = rootGetters.groupSettings?.distributionDate
+        if (!currentPeriod) { return false }
+
+        const nextPeriod = rootGetters.periodAfterPeriod(currentPeriod)
+        const now = new Date().toISOString()
+        const comparison = comparePeriodStamps(nextPeriod, now)
+
+        return rootGetters.ourGroupProfile?.incomeDetailsType === 'pledgeAmount' &&
+          (comparison > 0 && comparison < DAYS_MILLIS * 7) &&
+          (rootGetters.ourPayments && rootGetters.ourPayments.todo.length > 0) &&
+          !myNotificationHas(item => item.type === 'NEAR_DISTRIBUTION_END' && item.data.period === currentPeriod)
+      },
+      emit ({ rootState, rootGetters }) {
+        sbp('gi.notifications/emit', 'NEAR_DISTRIBUTION_END', {
+          createdDate: new Date().toISOString(),
+          groupID: rootState.currentGroupId,
+          period: rootGetters.groupSettings.distributionDate
+        })
+      },
+      shouldClearStateKey ({ rootGetters }) {
+        const currentPeriod = rootGetters.groupSettings.distributionDate
+        return rootGetters.currentNotifications.filter(item => item.type === 'NEAR_DISTRIBUTION_END')
+          .every(item => item.data.period !== currentPeriod)
+      }
+    }
+  },
+  {
+    type: PERIODIC_NOTIFICATION_TYPE.MIN1,
+    notificationData: {
+      stateKey: 'nextDistributionPeriod',
+      emitCondition ({ rootGetters }) {
+        if (!rootGetters.ourGroupProfile?.incomeDetailsType) return false // if income-details are not set yet, ignore.
+
+        const currentPeriod = rootGetters.groupSettings?.distributionDate
+        const now = new Date().toISOString()
+        let check = false
+
+        if (this.distPeriodRecords &&
+          comparePeriodStamps(this.distPeriodRecords.prevPeriod, this.distPeriodRecords.prevNow) > 0 &&
+          comparePeriodStamps(currentPeriod, now) < 0) { check = true }
+
+        if (this.distPeriodRecords && comparePeriodStamps(currentPeriod, this.distPeriodRecords.prevPeriod) > 0) { check = true }
+
+        this.distPeriodRecords = {
+          prevPeriod: currentPeriod,
+          prevNow: now
+        }
+
+        return check
+      },
+      emit ({ rootState, rootGetters }) {
+        sbp('gi.notifications/emit', 'NEW_DISTRIBUTION_PERIOD', {
```

Comment:
@taoeffect I have a question regarding this **'NEW_DISTRIBUTION_PERIOD'** notification here.
What the logic here does is, it checks every 1 minute if the distribution period has changed and send notification if it has, which means it's currently only targeting a very specific case where the distribution period changes **while the user is online**. (which happens a lot while we're testing/developing but I guess would rarely happens in real usage)

So do you think we have to add this same entry to the _one-time_ notifications list so that once the user logs in to the group, the app checks if the user has been notified for the currently-active distribution period and send the notification if that's not the case? :)

Reply:
I don't know, I'm thinking maybe there's a way to improve the emitCondition logic that hasn't been explored?

For example, what about simply checking for `comparePeriodStamps(dateToPeriodStamp(now), currentPeriod) > 0` ?

Btw - remember to convert `now` to a period stamp using `dateToPeriodStamp` before using it inside of `comparePeriodStamps`.

Reply:
@taoeffect Updated the logic in `emitCondition()` to simplify it.